frontend/src/app/main/data/workspace/selection.cljs

```diff
@@ -456,22 +456,32 @@
 
                 id-duplicated   (first new-ids)
 
-                frames (into #{}
-                             (map #(get-in objects [% :frame-id]))
-                             ids)
-                undo-id (js/Symbol)]
-
+                frames          (into #{}
+                                      (map #(get-in objects [% :frame-id]))
+                                      ids)
+
+                shapes          (map (d/getf objects) ids)
+                components      (filter ctk/instance-head? shapes)
+                create-event    #(ptk/event
+                                  ::ev/event
+                                  {::ev/name "use-library-component"
+                                   ::ev/origin "duplicate"
+                                   :external-library (not= file-id (:component-file %))})
+                events          (map create-event components)
+                undo-id         (js/Symbol)]
+            (rx/concat
+             (rx/from events)
```

Comment:
I think you don't need all this intermediate variables for declaring events, so this code can be writren in this manner:
```clojure
  (rx/concat
   (->> (map (d/getf objects) ids)
        (filter ctk/instance-head?)
        (map (fn [{:keys [component-file]}]
               (ptk/event ::ev/event
                          {::ev/name "use-library-component"
                           ::ev/origin "duplicate"
                           :external-library (not= file-id component-file)})))
        (rx/from))
   ...))
```

or 

```clojure
(rx/concat
 (->> (rx/from ids)
      (rx/map (d/getf objects))
      (rx/filter ctk/instance-head?)
      (rx/map (fn [{:keys [component-file]}]
                (ptk/event ::ev/event
                           {::ev/name "use-library-component"
                            ::ev/origin "duplicate"
                            :external-library (not= file-id component-file)}))))
 ...)
 ```

Reply:
Fixed